qiskit/transpiler/synthesis/permrowcol.py

```diff
@@ -0,0 +1,100 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Perm_row_col function implementation"""
+
+import numpy as np
+
+from qiskit.transpiler import CouplingMap
+from qiskit import QuantumRegister, QuantumCircuit
+
+
+class PermRowCol:
+    """Permrowcol algorithm"""
+
+    def __init__(self, coupling_map: CouplingMap):
+        self._coupling_map = coupling_map
+
+    def perm_row_col(self, parity_mat: np.ndarray, coupling_map: CouplingMap) -> QuantumCircuit:
+        """Run permrowcol algorithm on the given parity matrix
+
+        Args:
+            parity_mat (np.ndarray): parity matrix representing a circuit
+            coupling_map (CouplingMap): topology constraint
+
+        Returns:
+            QuantumCircuit: synthesized circuit
+        """
+        # TODO
+        circuit = QuantumCircuit(QuantumRegister(0))
+        return circuit
+
+    def choose_row(self, vertices: np.ndarray, parity_mat: np.ndarray) -> int:
```

Comment:
Nitpicky but it actually returns a numpy.int64

Reply:
Is it supposed to return numpy.int64? I'll change the tests etc. to better match these then.